### PR TITLE
feature(backend): add temporary buffer to subscription manager

### PR DESCRIPTION
**Pull request by mathnogueira**
This PR adds a temporary buffer to the subscription manager. This makes sure that if someone subscribes to a topic after an event is received, it will still receive those events.

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

### CODE REVIEW

File: server/subscription/buffer.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	b.messages = append(b.messages, message)
+	b.lastMessageTimestamp = time.Now()
+}
+
+func (b *TemporaryBuffer) GetMessages() []Message {
+	return b.messages
+}
+
+func (b *TemporaryBuffer) CleanUp() {
+	if len(b.messages) > 0 && time.Since(b.lastMessageTimestamp) > 10*time.Second {
```
Comment (danielbdias) [inline server/subscription/buffer.go:30]: Can we move this magic number `10 * time.Second` to a constant?

File: server/subscription/buffer.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
+package subscription
```
Comment (danielbdias) [inline server/subscription/buffer.go:1]: Brainstorm: does it make sense in the future to start to using queue engines to deal with event messaging?
I was thinking if we could use something like https://github.com/adjust/rmq or similar.

File: server/subscription/manager.go [hunk 3]
```diff
@@ -1,67 +1,45 @@
...
+	cleanupPeriod time.Duration
 }
 
 func NewManager() *Manager {
-	return &Manager{
-		subscriptions: make(map[string][]Subscriber),
-		mutex:         sync.Mutex{},
+	manager := &Manager{
+		subscriptions: NewBufferedSubscriptions(),
+		cleanupPeriod: 10 * time.Second,
```
Comment (danielbdias) [inline server/subscription/manager.go:13]: Can we move this config to constant?

**Review by danielbdias — APPROVED** [review]

**Review by jorgeepc — COMMENTED** [review]
Tested `tests` and `transactions` (they use the web socket too). Noticed a couple of things:
- It seems that the `clean up` process is not correct. I keep getting all the messages in the web socket even if I refresh the page.
- Transactions are failing in my local env. If I run a transaction the server stops sending web socket updates and the transactions stays in the `Executing` state forever.

File: server/subscription/buffer.go (hunk 2 above)
Comment (mathnogueira) [reply server/subscription/buffer.go:1]: I think it makes sense to close this PR and to work with some queue engine.

Comment (danielbdias) [reply server/subscription/buffer.go:1]: ok!

**Pull request closed without merging**